examples/aggregator/module/rust/src/lib.rs

```diff
@@ -160,10 +147,31 @@ impl Aggregator for AggregatorNode {
     }
 }
 
-oak::entrypoint!(grpc_worker => |in_channel| {
+oak::entrypoint!(aggregator => |in_channel| {
     oak::logger::init_default();
-    let dispatcher = AggregatorDispatcher::new(AggregatorNode::new());
-    oak::run_event_loop(dispatcher, oak::io::Receiver::<grpc::Invocation>::new(in_channel));
+
+    // Receive the initialization message.
+    let init_receiver = oak::io::Receiver::<AggregatorInit>::new(in_channel);
+    let init_message: AggregatorInit = init_receiver.receive().expect("Couldn't receive init message");
+    let grpc_server_invocation_receiver = init_message
+        .grpc_server_invocation_receiver
+        .expect("Couldn't receive gRPC invocation receiver")
+        .receiver
+        .expect("Empty gRPC invocation sender");
+    let grpc_client_invocation_sender = init_message
+        .grpc_client_invocation_sender
+        .expect("Couldn't receive gRPC invocation sender")
+        .sender
+        .expect("Empty gRPC invocation sender");
+
+    // Run event loop and handle incoming invocations.
+    let node = AggregatorNode::new(
+        oak::grpc::client::Client {
+            invocation_sender: grpc_client_invocation_sender
+        }
+    );
+    let dispatcher = AggregatorDispatcher::new(node);
+    oak::run_event_loop(dispatcher, grpc_server_invocation_receiver);
```

Comment:
@ipetr0v could you think (perhaps in a separate issue) of a better model to make it clear that there are essentially two `Node<T>` instances here, one that receives a single init message, and then from that, one that listens for incoming invocations. Perhaps the async work that @wildarch is doing will help with this in the future too.

Reply:
Technically there is only one Node, that has 2 incoming channels:
- One for reading the init message
- Another for reading incoming requests

Reply:
FYI https://github.com/project-oak/oak/pull/1572